lib/text/ui_text_displayer.js

```diff
@@ -190,9 +190,12 @@ shaka.text.UITextDisplayer = class {
       }
     });
 
+    // Calculate the video bounding box once before processing all the
+    // current cues, since this is an expensive operation.
+    const videoBoundingBox = this.videoContainer_.getBoundingClientRect();
```

Comment:
I'm worried that putting this here is still going to cause a performance bottleneck, since this line is called many times. Also, videoBoundingBox is only needed in a very very special case - when the cue has a region, a region id, and isn't nested or leafed.
Would it be best to leave it where it was before? Or should the entire logic be refactored so we don't need this bounding box.

Reply:
Seems fine with the check added above.  If there are no new cues to display, we don't compute the bounding box.  So the call is skipped between cue changes and when the display is off.